libs/logger/logger.hpp

```diff
@@ -12,20 +12,34 @@
 #include <numeric>  // for std::accumulate
 #include <string>
 
-/// Allows to log objects, which have toString() method without calling it, e.g.
-/// log.info("{}", myObject)
-template <typename StreamType, typename T>
-auto operator<<(StreamType &os, const T &object)
-    -> decltype(os << object.toString()) {
-  return os << object.toString();
-}
-
+#include <fmt/core.h>
 #include <fmt/format.h>
-#include <fmt/ostream.h>
 // Windows includes transitively included by format.h define interface as
 // struct, leading to compilation issues
 #undef interface
 
+namespace fmt {
+  /// Allows to log objects, which have toString() method without calling it,
+  /// e.g. log.info("{}", myObject)
+  template <typename T>
+  struct formatter<
+      T,
+      std::enable_if_t<std::is_same<decltype(std::declval<T>().toString()),
```

Comment:
toString() may also return const&, or even string_view. Let's use std::decay<>
```
std::is_same_v<std::decay_t<decltype(std::declval<T>().toString())>,std::string>
```

Reply:
I've checked and changing the code above into:
```
  template <typename T>
  struct formatter<
      T,
      std::enable_if_t<std::is_same<std::decay_t<decltype(std::declval<T>().toString())>,
```
does not add any compilation error (I've checked with tests).